File: project_6/readability.py
from cs50 import get_string, get_float, get_int

text = get_string("Text: ")
#print(f"Text: {text}")
#print(f"Lenght of the text: {len(text)}")

count_letters = 0 #counter for letters
#last_char = text[-1]

for i in range(0, len(text)):
    #print(text[i])
    if text[i].islower() or text[i].isupper(): #characters contain lower or upper letter
        count_letters += 1
        #print(f"Yes letter {text[i]}") #just for checking
#print(f"Number of letters: {count_letters}")


count_words = len(text.split()) #text is split into words
#print(f"Number of words: {count_words}")


count_sentences = 0

for j in range(0, len(text)):
    if text[j] == '.' or text[j] == '!' or text[j] == '?': #characters contain . ! ?
        count_sentences += 1
#print(f"Number of sentences: {count_sentences}")

L = float(count_letters / count_words) * 100
#print(f"L is: {L}")

S = float(count_sentences / count_words) * 100
#print(f"S is: {S}")


index = 0.0588 * L - 0.296 * S - 15.8
#print(f"Index is: {index}")

grade = round(index)
#print(f"Grade is: {grade}")

if grade >= 16:
    print("Grade 16+")
elif grade < 1:
    print("Before Grade 1")
else:
    print(f"Grade {grade}")

File: project_2/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>



int main(void)
{
    string s = get_string("Text: ");
    printf("%s\n", s);

    int countl = 0; //counter for letters
    int i = 0;

    for (i = 0; i < strlen(s); i++)
    {
        if (islower(s[i]) || isupper(s[i])) //characters contain lower or upper letter
        {
            countl++;
            //printf("Yes letter %c\n", s[i]); just for checking
        }
    }
    //printf("%i letter(s)\n", countl);

    int countw = 0; //counter for words
    int j = 0;

    for (j = 0; j <= strlen(s); j++)
    {
        if (s[j] == '\0' || isspace(s[j])) //characters contain \0 or space
        {
            countw++;
            //printf("Yes space  %c\n", s[j]); just for checking
        }
    }
    //printf("%i word(s)\n", countw);

    int counts = 0; //counter for sentences
    int k = 0;

    for (k = 0; k <= strlen(s); k++)
    {
        if (s[k] == '.' || s[k] == '!' || s[k] == '?') //characters contain . ! ?
        {
            counts++;
            //printf("Yes %c\n", s[k]); just for checking
        }
    }
    //printf("%i sentence(s)\n", counts);

    float L = ((float) countl / (float) countw) * 100;
    //printf("%.2f\n", L);

    float S = ((float) counts / (float) countw) * 100;
    //printf("%.2f\n", S);



    float index = 0.0588 * L - 0.296 * S - 15.8;
    //printf("Index is %f\n", index);

    int grade = round(index);


    if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}